eox_tenant/monkey_patch/monkey_patch_proxys.py

```diff
@@ -0,0 +1,160 @@
+"""
+Monkey patch proxys that allows to override the platform models.
+"""
+import logging
+import json
+from itertools import chain
+
+import six
+from django.conf import settings
+from django.core.cache import cache
+
+from eox_tenant.edxapp_wrapper.site_configuration_module import get_site_configuration_models
+from eox_tenant.models import TenantConfig, Microsite
+
+SiteConfigurationModels = get_site_configuration_models()
+TENANT_ALL_ORGS_CACHE_KEY = "tenant.all_orgs_list"
+EOX_TENANT_CACHE_KEY_TIMEOUT = getattr(
+    settings,
+    "EOX_TENANT__CACHE_KEY_TIMEOUT",
+    300
+)
+TENANT_MICROSITES_ITERATOR_KEY = "tenant-microsites-iterator"
+logger = logging.getLogger(__name__)
+
+
+class TenantSiteConfigProxy(SiteConfigurationModels.SiteConfiguration):
+    """
+    This a is Proxy model for SiteConfiguration from <openedx.core.djangoapps.site_configuration.models>.
+    This allows to add or override methods using as base the SiteConfiguration model.
+    More information in https://docs.djangoproject.com/en/3.0/topics/db/models/#proxy-models
+    """
+
+    class Meta:
+        """ Set as a proxy model. """
+        proxy = True
+
+    def __unicode__(self):
+        key = getattr(settings, "EDNX_TENANT_KEY", "No tenant is active at the moment")
+        return u"<Tenant proxy as site_configuration: {}>".format(key)
+
+    @property
+    def enabled(self):
```

Comment:
This method is most certainly not doing what it should be doing.

USE_EOX_TENANT is a global flag to turn the complete plugin on and off.

The real logic for this is to check for the existance of the EDNX_TENANT_KEY in the settings object

```
    def enabled(self):
        if getattr(settings, 'EDNX_TENANT_KEY', None):
            return True
        return False
```

Reply:
I think that with this change, the AvailableScreen Middleware should work as expected